homeassistant/components/homeassistant/exposed_entities.py

```diff
@@ -442,25 +444,41 @@ def ws_list_exposed_entities(
     result: dict[str, Any] = {}
 
     exposed_entities = hass.data[DATA_EXPOSED_ENTITIES]
-    required_assistant = msg.get("assistant")
     entity_registry = er.async_get(hass)
     for entity_id in chain(exposed_entities.entities, entity_registry.entities):
-        entity_settings = async_get_entity_settings(hass, entity_id)
-        if required_assistant and (
-            (required_assistant not in entity_settings)
-            or (not entity_settings[required_assistant].get("should_expose"))
-        ):
-            # Not exposed to required assistant
-            continue
-
         result[entity_id] = {}
+        entity_settings = async_get_entity_settings(hass, entity_id)
         for assistant, settings in entity_settings.items():
             if "should_expose" not in settings:
                 continue
             result[entity_id][assistant] = settings["should_expose"]
     connection.send_result(msg["id"], {"exposed_entities": result})
 
 
+@callback
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "homeassistant/expose_entity/list_exposed",
+        vol.Required("assistant"): vol.In(KNOWN_ASSISTANTS),
+    }
+)
+def ws_list_entities_exposed_to_assistant(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """List entities which are exposed to an assistant."""
+    exposed_entities = hass.data[DATA_EXPOSED_ENTITIES]
+    assistant = msg.get("assistant")
+    entity_registry = er.async_get(hass)
+    result = [
+        entity_id
+        for entity_id in chain(exposed_entities.entities, entity_registry.entities)
+        if assistant in (entity_settings := async_get_entity_settings(hass, entity_id))
+        and entity_settings[assistant]["should_expose"]
```

Comment:
Is "should_expose" guaranteed to be in the settings? The original command seemed more cautious for some reason.

Reply:
You're right, I'll change it 👍